intel_extension_for_transformers/llm/runtime/graph/application/main_pybind.cpp

```diff
@@ -0,0 +1,279 @@
+
+    if (next_token_id == model_token_eos()) {
+      break;
+    }
+  }
+
+  return ret;
+}
+
+std::string Model::generate(const std::string& prompt, bool stream_mode) {
+    if (stream_mode) {
+        return generate_one_token(prompt);
+    }
+    
+    return generate_tokens(prompt);
+}
+
+int Model::post_process(float* logits) {
+  int alpha_frequency = 0;
+  int alpha_presence = 0;
+  int repeat_last_n = 64;
+  float repeat_penalty = 1.10f;
+  int top_k = 40;
+  float tfs_z = 1.00f;
+  float typical_p = 1.00f;
+  float top_p = 0.95f;
+  float temp = 0.80f;
+  std::vector<model_token_data> candidates;
+  candidates.reserve(n_vocab);
+  for (model_token token_id = 0; token_id < n_vocab; token_id++) {
+    candidates.emplace_back(model_token_data{token_id, logits[token_id], 0.0f});
+  }
+  model_token_data_array candidates_p = {candidates.data(), candidates.size(), false};
+  
+  // Apply penalties
+  float nl_logit = logits[model_token_nl()];
+  auto last_n_repeat = std::min(std::min((int)last_n_tokens.size(), repeat_last_n), n_ctx);
+  model_sample_repetition_penalty(ctx, &candidates_p, last_n_tokens.data() + last_n_tokens.size() - last_n_repeat,
+                                  last_n_repeat, repeat_penalty);
+  model_sample_frequency_and_presence_penalties(ctx, &candidates_p,
+                                                last_n_tokens.data() + last_n_tokens.size() - last_n_repeat,
+                                                last_n_repeat, alpha_frequency, alpha_presence);
+  // int id = model_sample_token_greedy(ctx, &candidates_p);
+  // Temperature sampling
+  model_sample_top_k(ctx, &candidates_p, top_k, 1);
+  model_sample_tail_free(ctx, &candidates_p, tfs_z, 1);
+  model_sample_typical(ctx, &candidates_p, typical_p, 1);
+  model_sample_top_p(ctx, &candidates_p, top_p, 1);
+  model_sample_temperature(ctx, &candidates_p, temp);
+  int id = model_sample_token(ctx, &candidates_p);
+  return id;
+}
+
+namespace py = pybind11;
+
+#if MODEL_NAME_ID == 1
+
+PYBIND11_MODULE(gptj_cpp, m)
+
+#elif MODEL_NAME_ID == 2
+
+PYBIND11_MODULE(falcon_cpp, m)
+
+#elif MODEL_NAME_ID == 3
+
+PYBIND11_MODULE(gptneox_cpp, m)
+
+#elif MODEL_NAME_ID == 4
+
+PYBIND11_MODULE(dolly_cpp, m)
+
+#elif MODEL_NAME_ID == 5
+
+PYBIND11_MODULE(llama_cpp, m)
+
+#elif MODEL_NAME_ID == 6
+
+PYBIND11_MODULE(mpt_cpp, m)
+
+#elif MODEL_NAME_ID == 7
+
+PYBIND11_MODULE(starcoder_cpp, m)
+
+#elif MODEL_NAME_ID == 8
+
+PYBIND11_MODULE(opt_cpp, m)
+
+#elif MODEL_NAME_ID == 9
+
+PYBIND11_MODULE(bloom_cpp, m)
+
+#elif MODEL_NAME_ID == 10
+
+PYBIND11_MODULE(chatglm2_cpp, m)
+
+#endif
+{
+    m.doc() = "cpp model python binding";
+    py::class_<Model>(m, "Model", py::module_local())
+        .def(py::init())
+        .def("init_model", &Model::init_model)
+        .def("generate", &Model::generate)
+        .def("set_n_predict", &Model::set_n_predict)
+        .def("set_batch_size", &Model::set_batch_size)
+        .def("set_ctx_size", &Model::set_ctx_size)
```

Comment:
how about get rid of set 
like n_predict     batch_size  ? seems also ok

Reply:
updated like this:
```
m.init_model(model_path = "ne_mpt_q4_j.bin",
             n_predict = 100,
             seed = 12,
             threads = 28)
```